src/core/Akka.API.Tests/CoreAPISpec.ApproveClusterSharding.approved.txt

```diff
@@ -156,70 +179,42 @@ namespace Akka.Cluster.Sharding
-        public bool GracefulShutdownInProgress { get; }
-        public Akka.Event.ILoggingAdapter Log { get; }
-        protected object RegistrationMessage { get; }
+        public ShardRegion(string typeName, System.Func<string, Akka.Actor.Props> entityProps, Akka.Cluster.Sharding.ClusterShardingSettings settings, string coordinatorPath, Akka.Cluster.Sharding.ExtractEntityId extractEntityId, Akka.Cluster.Sharding.ExtractShardId extractShardId, object handOffStopMessage, Akka.Cluster.Sharding.Internal.IRememberEntitiesProvider rememberEntitiesProvider) { }
         public Akka.Actor.ITimerScheduler Timers { get; set; }
-        public int TotalBufferSize { get; }
-        protected bool MatchingRole(Akka.Cluster.Member member) { }
         protected override void PostStop() { }
         protected override void PreStart() { }
         protected override bool Receive(object message) { }
-        public sealed class StartEntity : Akka.Cluster.Sharding.IClusterShardingSerializable
+        public sealed class StartEntity : Akka.Cluster.Sharding.IClusterShardingSerializable, System.IEquatable<Akka.Cluster.Sharding.ShardRegion.StartEntity>
         {
             public readonly string EntityId;
             public StartEntity(string entityId) { }
             public override bool Equals(object obj) { }
+            public bool Equals(Akka.Cluster.Sharding.ShardRegion.StartEntity other) { }
             public override int GetHashCode() { }
-        }
-        public sealed class StartEntityAck : Akka.Cluster.Sharding.IClusterShardingSerializable, Akka.Event.IDeadLetterSuppression
-        {
-            public readonly string EntityId;
-            public readonly string ShardId;
-            public StartEntityAck(string entityId, string shardId) { }
-            public override bool Equals(object obj) { }
-            public override int GetHashCode() { }
+            public override string ToString() { }
         }
     }
     public sealed class ShardRegionStats : Akka.Cluster.Sharding.IClusterShardingSerializable, System.IEquatable<Akka.Cluster.Sharding.ShardRegionStats>
     {
+        public readonly System.Collections.Immutable.IImmutableSet<string> Failed;
         public readonly System.Collections.Immutable.IImmutableDictionary<string, int> Stats;
-        public ShardRegionStats(System.Collections.Immutable.IImmutableDictionary<string, int> stats) { }
-        public bool Equals(Akka.Cluster.Sharding.ShardRegionStats other) { }
+        public ShardRegionStats(System.Collections.Immutable.IImmutableDictionary<string, int> stats, System.Collections.Immutable.IImmutableSet<string> failed) { }
```

Comment:
Avoid changing an existing public constructor signature, this should be made into a constructor overload.

Reply:
added